tests/set-namespace/ensure-name-substring/Kptfile

```diff
@@ -0,0 +1,12 @@
+apiVersion: kpt.dev/v1
+kind: Kptfile
+metadata:
+  name: example
+pipeline:
+  mutators:
+    - image: gcr.io/kpt-fn/ensure-name-substring:unstable
```

Comment:
What are we trying to do here?
Are we trying to ensure they both update the `ServiceAccount` under `subjects`?

Reply:
This was in response to [this discussion](https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/680#discussion_r769086646) with @droot, i.e.
>it will be good to have a test case that involves ensure-name-substring function in the pipeline and ensure this change works well with that too.

My Interpretation of the idea is as you say, to ensure they both update the `ServiceAccount` under `subjects`

Reply:
Got it